lib/luahid.c

```diff
@@ -99,14 +103,50 @@ do { 						\
 	lua_setfield(L, idx - 1, #field);	\
 } while (0)
 
-static inline void luahid_pushdevid(lua_State *L, int idx, const struct hid_device_id *device_id)
+#define luahid_pcall(L, func, idx, arg) 			\
+do { 								\
+	int n = lua_gettop(L); 					\
+	lua_pushcfunction(L, func); 				\
+	lua_pushvalue(L, idx - 1); 				\
+	lua_pushlightuserdata(L, (void *)arg); 			\
+	if (lua_pcall(L, 2, LUA_MULTRET, 0) != LUA_OK) { 	\
+		pr_err("%s: %s\n", #func, lua_tostring(L, -1));	\
+		lua_settop(L, n);				\
+	} 							\
+} while (0)
+
+static inline int luahid_pushdevid(lua_State *L)
 {
+	struct hid_device_id *device_id = (struct hid_device_id *)lua_touserdata(L, 2);
 	lua_newtable(L);
 	luahid_setfield(L, -1, device_id, bus);
 	luahid_setfield(L, -1, device_id, group);
 	luahid_setfield(L, -1, device_id, vendor);
 	luahid_setfield(L, -1, device_id, product);
```

Comment:
this entire block can be moved to a macro.. you have the exact same pattern in pushdev()